### ISSUE TITLE
How should we handle directory renames with 'getEditsForFileRename'?

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#24260]
For a project:

```
jsconfig.json
index.ts
sub/
    a.ts
    b.ts
```
 
Where `index.ts`:

```
import {a} from './sub/a'
import {b} from './sub/b'
console.log(a, b);
```

If the user renames `sub` -> `sub2`, how should we handle updating the paths using the `getEditsForFileRename` api? Is this something we want to support in V1?

I'm concerned that renaming directories may require a  call to `getEditsForFileRename` for each file, which will be noisy

**Comment by andy-ms** [comment]
The code doesn't support this currently, so this is something we'll have to add.

**Comment by mjbvz** [comment]
Ok, should the API be updated so that we can support this in the future? Or do you think the existing API will work?

**Comment by andy-ms** [comment]
I think we can just use the current function and detect that the path passed in is a directory.

**Comment by mjbvz** [comment]
Good with me. Just one question: what would the `file` be in that case? Just any file ts in the current project?

**Comment by andy-ms** [comment]
Right. The 'file' argument doesn't even show up in the `LanguageService` method.

**Comment by mhegazy** [comment]
@andy-ms let's try to get this in the next servicing release 2.9.2
